SchedulerBot/SchedulerBot.Business.Commands/ManageCommand.cs

```diff
@@ -0,0 +1,137 @@
+	public class ManageCommand : BotCommand
+	{
+		private readonly SchedulerBotContext context;
+		private readonly IRandomByteGenerator randomByteGenerator;
+		private readonly TimeSpan linkExpirationPeriod;
+		private readonly int linkIdLength;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="ManageCommand"/> class.
+		/// </summary>
+		/// <param name="context">The context.</param>
+		/// <param name="randomByteGenerator">The random byte generator.</param>
+		/// <param name="configuration">The configuration.</param>
+		/// <param name="logger">The logger.</param>
+		public ManageCommand(
+			SchedulerBotContext context,
+			IRandomByteGenerator randomByteGenerator,
+			IConfiguration configuration,
+			ILogger<ManageCommand> logger) : base("manage", logger)
+		{
+			this.context = context;
+			this.randomByteGenerator = randomByteGenerator;
+
+			linkExpirationPeriod = TimeSpan.Parse(configuration["Commands:Manage:LinkExpirationPeriod"], CultureInfo.InvariantCulture);
+			linkIdLength = int.Parse(configuration["Commands:Manage:LinkIdLength"], CultureInfo.InvariantCulture);
+		}
+
+		/// <inheritdoc />
+		protected override async Task<CommandExecutionResult> ExecuteCoreAsync(Activity activity, string arguments)
+		{
+			Logger.LogInformation(
+				"Creating a manage link for channel '{0}' and conversation '{1}'",
+				activity.ChannelId,
+				activity.Conversation.Id);
+
+			ManageConversationLink manageConversationLink = CreateManageConversationLink(activity);
+
+			await context.ManageConversationLinks.AddAsync(manageConversationLink);
+			await context.SaveChangesAsync();
+
+			Logger.LogInformation(
+				"Created the manage link '{0}' for channel '{1}' and conversation '{2}'",
+				manageConversationLink.Text,
+				activity.ChannelId,
+				activity.Conversation.Id);
+
+			return CommandExecutionResult.Success(manageConversationLink.Text);
+		}
+
+		private ManageConversationLink CreateManageConversationLink(Activity activity)
+		{
+			string linkText = GenerateRandomString(linkIdLength);
+			DateTime linkCreationTime = DateTime.UtcNow;
+			DateTime linkExpirationTime = linkCreationTime + linkExpirationPeriod;
+
+			return new ManageConversationLink
+			{
+				ChannelId = activity.ChannelId,
+				ConversationId = activity.Conversation.Id,
+				Text = linkText,
+				CreatedOn = linkCreationTime,
+				ExpiresOn = linkExpirationTime
+			};
+		}
+
+		private string GenerateRandomString(int length)
+		{
+			string randomBase64String = GetRandomBase64String(length);
+			string randomUrlEncodedString = UrlEncodeBase64String(randomBase64String, length);
+
+			return randomUrlEncodedString;
+		}
+
+		private string GetRandomBase64String(int minLength)
+		{
+			byte[] randomBytes = randomByteGenerator.Generate(minLength);
+			string randomBase64String = Convert.ToBase64String(randomBytes);
+
+			return randomBase64String;
+		}
+
+		private static string UrlEncodeBase64String(string base64String, int length)
```

Comment:
As possible option here can be https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.webutilities.webencoders.base64urlencode?view=aspnetcore-2.0 but for this place, it will be ok to have custom implementation

Reply:
Just didn't want to reference `Microsoft.AspNetCore.WebUtilities.dll` in this project, especially that the algorithm is pretty simple.